src/jquery.multiselect.js

```diff
@@ -1280,22 +1271,54 @@
     },
 
     checkAll: function(e) {
-      this._toggleChecked(true);
-      this._trigger('checkAll');
+      this._trigger('beforeCheckAll');
+
+      var maxSelected = this.options.maxSelected;
+      if (maxSelected === null || maxSelected > this.$inputs.length) {
```

Comment:
I think it might be better to set the `checkAllText` option to `null` if `maxSelected` is set. That way the user just can't press it. It would be a clearer interface for the user interacting with the widget.